### PR TITLE
chore: ensure correct artifact building

### PR DESCRIPTION
**Pull request by matiasbargas**

### CODE REVIEW

File: scripts/prepareDist.ts [hunk 1]
```diff
@@ -45,19 +48,26 @@ async function copyIndex(filename: string) {
...
   {
     let content = readFileSync(targetIndexHtml).toString()
 
-    if (!content.includes(`${filename}.js`)) {
-      throw new Error(`index.html is dirty and does\'t contain the text "${filename}.js"`)
-    }
+    let didReplace = false
+
+    content = content.replace(new RegExp(fileNameToReplace + '.(S+.)?js'), function(original) {
+      console.log(`>   replacing ${original} -> ${newFileName}`)
```
Comment (fmiras) [inline scripts/prepareDist.ts]:
```suggestion
      console.log(`> replacing ${original} -> ${newFileName}`)
```

File: scripts/prepareDist.ts [hunk 2]
```diff
@@ -45,19 +48,26 @@ async function copyIndex(filename: string) {
     throw new Error(`${targetIndexHtml} does not exist`)
   }
 
-  console.log(`> replace ${filename}.js -> ${newFileName} in html`)
+  console.log(`> replace ${fileNameToReplace}.js -> ${newFileName} in html`)
```
Comment (fmiras) [inline scripts/prepareDist.ts:51]: Please be consistent using present continuous verbs or infinitive, but not the two of them

File: scripts/prepareDist.ts [hunk 3]
```diff
@@ -45,19 +48,26 @@ async function copyIndex(filename: string) {
...
   {
     let content = readFileSync(targetIndexHtml).toString()
 
-    if (!content.includes(`${filename}.js`)) {
-      throw new Error(`index.html is dirty and does\'t contain the text "${filename}.js"`)
-    }
+    let didReplace = false
+
+    content = content.replace(new RegExp(fileNameToReplace + '.(S+.)?js'), function(original) {
```
Comment (fmiras) [inline scripts/prepareDist.ts:57]: Use arrow function here

Comment (fmiras) [inline scripts/prepareDist.ts:57]: I think we should use regex literals to be consistent with the rest of the code

File: scripts/prepareDist.ts [hunk 4]
```diff
@@ -45,19 +48,26 @@ async function copyIndex(filename: string) {
...
 
-  console.log(`> replace ${filename}.js -> ${newFileName} in html`)
+  console.log(`> replace ${fileNameToReplace}.js -> ${newFileName} in html`)
   {
     let content = readFileSync(targetIndexHtml).toString()
 
-    if (!content.includes(`${filename}.js`)) {
-      throw new Error(`index.html is dirty and does\'t contain the text "${filename}.js"`)
-    }
+    let didReplace = false
```
Comment (fmiras) [inline scripts/prepareDist.ts:55]: Just compare the already existing variables, you can use `content` for "original" and `result` for the regex match, and if they are the same you already know if it was replaced or not

**Review by fmiras — CHANGES_REQUESTED** [review]

File: scripts/prepareDist.ts (hunk 3 above)
Comment (eordano) [reply scripts/prepareDist.ts:57]: I wasn't aware that regexp literals was possible with template `` strings, is it?

Comment (fmiras) [reply scripts/prepareDist.ts:57]: Just looked into it, you can't so my bad!

**Pull request closed without merging**